scripts/verify_base_environment.sh

```
@@ -0,0 +1,67 @@
+    # Check udev socket
+    echo -e "\n2. udev socket:"
+    ls -la /run/udev/control 2>/dev/null || echo "/run/udev/control not found"
+    
+    # Check existing USB devices
+    echo -e "\n3. Existing USB devices:"
+    ls -la /dev/ttyUSB* 2>/dev/null || echo "No USB serial devices"
+    ls -la /dev/ttyACM* 2>/dev/null || echo "No ACM devices"
+    
+    # Check device subsystems
+    echo -e "\n4. TTY subsystem devices:"
+    ls /sys/class/tty/ | grep -E "(USB|ACM)" || echo "No USB TTY devices in sysfs"
```

Shellcheck (0.10.0)

[warning] Don't use ls | grep. Use a glob or a for loop with a condition to allow non-alphanumeric filenames.

(SC2010)